File: 43 - Find the Missing Number in Consecutive Array/solution.js
function findMissingNumber(arr) {
  const n = arr.length + 1;
  const expectedSum = (n * (n + 1)) / 2;
  const actualSum = arr.reduce((acc, cur) => acc + cur, 0);
  return expectedSum - actualSum;
}

console.log(findMissingNumber([1, 2, 3, 4, 5, 6, 7])); // 3

File: 43 - Find the Missing Number in Consecutive Array/solution.java

public class solution {

    public static int findMissingNumber(int[] arr) {
        int n = arr.length + 1;
        int expectedSum = (n * (n + 1)) / 2;
        int actualSum = 0;
        for (int i = 0; i < arr.length; i++) {
            actualSum += arr[i];
        }
        return expectedSum - actualSum;
    }

    public static void main(String[] args) {
        int[] arr = { 1, 2, 3, 4, 5, 6, 7 };
        System.out.println(findMissingNumber(arr));
    }
}
